### ISSUE TITLE
Authorization object is missing "wildcard" field when issuing wildcard certificates

### ISSUE THREAD

**Issue by sipe90** [grindsa/acme2certifier#76]
I'm having issues with cert-manager when issuing wildcard certificates using DNS01 challenges. Cert-manager presents the challenge incorrectly because the authorization object does not contain the wildcard field with value true.

According to the ACME spec:
```
wildcard (optional, boolean):  This field MUST be present and true
      for authorizations created as a result of a newOrder request
      containing a DNS identifier with a value that was a wildcard
      domain name.  For other authorizations, it MUST be absent.
```

So when trying to issue a wildcard certificate for `*.example.com`, cert-manager sees that the wildcard-field is missing from the authorization object and so assumes that the certificate is NOT a wildcard certificate. This results in cert-manager creating a TXT record of `_acme-challenge.*.example.com` instead of `_acme-challenge.example.com` and so the challenge validation never succeeds.

**Comment by grindsa** [comment]
Thx for spotting this. Seems cert-manager is the first acme-client make use of this field.  Creating a fix should not be a big issue. Will look into this later this week...

**Comment by grindsa** [comment]
A fix got already included in the devel branch and a docker image based on apache2-wsgi got uploaded to [docker-hub](https://hub.docker.com/layers/grindsa/acme2certifier/devel/images/sha256-6693802783af98be5e4880f3a0e9b22b92c8fc2f0bfd0b11617f82eb91b3930d?context=explore). Can you please give it a try (`docker pull grindsa/acme2certifier:devel`) and check if it works for you? I am sorry but I cannot test enrolment of wildcard-certificates via cert-manager as I am missing the environment to do so.

**Comment by sipe90** [comment]
I'm getting a deserialization error from cert-manager: `E1115 07:03:09.340176       1 controller.go:163] cert-manager/controller/orders "msg"="re-queuing item due to error processing" "error"="acme: invalid response: json: cannot unmarshal string into Go struct field wireAuthz.Wildcard of type bool" "key"="cert-manager/xxxxxxxxx"`. The wildcard field value seems to be returned as a string while it is expected to be boolean.

**Comment by grindsa** [comment]
New devel build got uploaded to dockerhub. Pls try again...

**Comment by sipe90** [comment]
Okay, now the parsing succees, but the created TXT record is still incorrect: `_acme-challenge.*.example.com`. The spec.wildcard field value is now true in the Challenge resource, but now I noticed this documentation about field spec.dnsName:
```
dnsName is the identifier that this challenge is for, e.g. example.com. If the requested DNSName is a 'wildcard', this field MUST be set to the non-wildcard domain, e.g. for `*.example.com`, it must be `example.com`.
```
Somehow the value still ended up being set to `*.example.com`.

**Comment by grindsa** [comment]
Ok. this explains why regression failed last night as the tests using  `acme.sh` did fail as well. The fix is already in devel. I will upload a new image during next hour...

**Comment by sipe90** [comment]
I tested with the latest version and now the certificate was issued successfully. Seems to be working as expected by cert-manager. Thank you!

**Comment by grindsa** [comment]
Fixes got included in [v0.19](https://github.com/grindsa/acme2certifier/releases/tag/0.19). Thus, I am closing the issue.
